src/target/sam4l.c

```diff
@@ -32,7 +32,7 @@
 /*
  * Flash Controller defines
  */
-#define FLASHCALW_BASE				0x400A0000
+#define FLASHCALW_BASE				0x400A0000L
```

Comment:
Rather than suffixing L here, could you use U please (this is used in an unsigned context, so should be marked as such)

Reply:
Unfortunately, U doesn't work:

```
C:/Users/Sean/esp/esp-idf/components/log/include/esp_log.h:270:27: note: in expansion of macro 'LOG_COLOR'
  270 | #define LOG_COLOR_I       LOG_COLOR(LOG_COLOR_GREEN)
      |                           ^~~~~~~~~
C:/Users/Sean/esp/esp-idf/components/log/include/esp_log.h:282:37: note: in expansion of macro 'LOG_COLOR_I'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                     ^~~~~~~~~~
C:/Users/Sean/esp/esp-idf/components/log/include/esp_log.h:414:86: note: in expansion of macro 'LOG_FORMAT'
  414 |         else                                { esp_log_write(ESP_LOG_INFO,       tag, LOG_FORMAT(I, format), esp_log_timestamp(), tag, ##__VA_ARGS__); } \
      |                                                                                      ^~~~~~~~~~
C:/Users/Sean/esp/esp-idf/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
C:/Users/Sean/esp/esp-idf/components/log/include/esp_log.h:340:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  340 | #define ESP_LOGE( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_ERROR,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
E:/Code/Farpatch/format-demo/main/main.c:63:5: note: in expansion of macro 'ESP_LOGE'
   63 |     ESP_LOGE("foo", "Some value: %"PRIx32, 0x90800000U);
      |     ^~~~~~~~
cc1.exe: some warnings being treated as errors
```

Is it alright if I use `UL`?

Reply:
in that case, could you use `UINT32_C()` around the constant - that'll make it unsigned and suffix it according to platform requirements which is more correct than 'UL'

Reply:
Alright, I removed the `U` suffix and wrapped the debug statement argument in `UINT32_C()`.

Reply:
It seems that `UINT32_C()` doesn't work on hosted mode. Hmm...